web-bundle/src/main/java/com/graphhopper/resources/MapboxResponseConverter.java

```diff
@@ -0,0 +1,324 @@
+    private static final TranslationMap trMap = new TranslationMap().doImport();
+
+
+    /**
+     * Converts a GHResponse into Mapbox compatible json
+     */
+    public static ObjectNode convertFromGHResponse(GHResponse ghResponse, Locale locale) {
+        ObjectNode json = JsonNodeFactory.instance.objectNode();
+
+        if (ghResponse.hasErrors())
+            throw new IllegalStateException("Cannot convert an errornous ghResponse");
+
+        PointList waypoints = ghResponse.getBest().getWaypoints();
+
+        final ArrayNode routesJson = json.putArray("routes");
+
+        List<PathWrapper> paths = ghResponse.getAll();
+
+        for (int i = 0; i < paths.size(); i++) {
+            PathWrapper path = paths.get(i);
+            ObjectNode pathJson = routesJson.addObject();
+
+            pathJson.put("geometry", WebHelper.encodePolyline(path.getPoints(), false, 1e6));
+            ArrayNode legsJson = pathJson.putArray("legs");
+
+            //TODO: support multiple legs (=> multiple waypoints...)
+            ObjectNode legJson = legsJson.addObject();
+
+            // TODO: Improve path descriptions, so that every path has a description, not just alternative routes
+            String summary;
+            if (!path.getDescription().isEmpty())
+                summary = String.join(",", path.getDescription());
+            else
+                summary = "GraphHopper Route " + i;
+            legJson.put("summary", summary);
+
+            // Copied from below
+            legJson.put("weight", path.getRouteWeight());
+            legJson.put("duration", convertToSeconds(path.getTime()));
+            legJson.put("distance", Math.round(path.getDistance()));
+
+            ArrayNode steps = legJson.putArray("steps");
+            InstructionList instructions = path.getInstructions();
+
+            for (int j = 0; j < instructions.size(); j++) {
+                ObjectNode instructionJson = steps.addObject();
+                putInstruction(instructions, j, locale, instructionJson);
+            }
+
+            pathJson.put("weight_name", "routability");
+            pathJson.put("weight", path.getRouteWeight());
+            pathJson.put("duration", convertToSeconds(path.getTime()));
+            pathJson.put("distance", Math.round(path.getDistance()));
+            pathJson.put("voiceLocale", locale.toLanguageTag());
+        }
+
+        final ArrayNode waypointsJson = json.putArray("waypoints");
+        for (int i = 0; i < waypoints.size(); i++) {
+            ObjectNode waypointJson = waypointsJson.addObject();
+            // TODO get names
+            waypointJson.put("name", "");
+            putLocation(waypoints.getLat(i), waypoints.getLon(i), waypointJson);
+        }
+
+        json.put("code", "Ok");
+        // TODO: Maybe we need a different format... uuid: "cji4ja4f8004o6xrsta8w4p4h"
+        json.put("uuid", UUID.randomUUID().toString().replaceAll("-", ""));
```

Comment:
What is the purpose of this uuid? Maybe we have to hash the response somehow for error reporting?

Reply:
The purpose is not documented (or I could not find it). I haven't checked how it is used in the code. I think this is used internally to match error reports with requests. I tried sending multiple requests with the same parameters, and the UUID was different every time (but only at the end, the first characters were similar, so there seems to be some logic behind this).